Source/SwiftLintFramework/Rules/WeakDelegateRule.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  WeakDelegate.swift
+//  SwiftLint
+//
+//  Created by Olivier Halligon on 11/08/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct WeakDelegateRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "weak_delegate",
+        name: "Weak Delegate",
+        description: "Delegates should be weak to avoid reference cycles.",
+        nonTriggeringExamples: [
+            "class Foo {\n  weak var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  weak var someDelegate: SomeDelegateProtocol?\n}\n",
+            "class Foo {\n  weak var delegateScroll: ScrollDelegate?\n}\n",
+            // We only consider properties to be a delegate if it has "delegate" in its name
+            "class Foo {\n  var scrollHandler: ScrollDelegate?\n}\n",
+            // Only trigger on instance variables, not local variables
+            "func foo() {\n  var delegate: SomeDelegate\n}\n"
+        ],
+        triggeringExamples: [
+            "class Foo {\n  var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  var scrollDelegate: ScrollDelegate?\n}\n",
+            "class Foo {\n  var delegateScroll: ScrollDelegate?\n}\n",
+        ]
+    )
+
+    public func validateFile(file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .VarInstance else {
+            return []
+        }
+
+        // Check if name contains "delegate"
+        let name = (dictionary["key.name"] as? String) ?? ""
+        let isDelegate = name.lowercaseString.rangeOfString("delegate") != nil
+        guard isDelegate else { return [] }
+
+        // Check if non-weak
+        let attributes = (dictionary["key.attributes"] as? [SourceKitRepresentable])?
```

Comment:
No a RegEx isn't enough:

1) The RegEx like the one you mentioned would need a negative look-behind expression to ensure the weak keyword isn't present (and this generally means non-optimal CPU-intensive backtracking in the RegEx engine)
2) Not sure if SwiftLint easily allows to match patterns against a whole line…while still being able to limit the matching to instance variable declarations only and not for example local variable?
3) The fact that the declaration is optional and a protocol isn't enough condition. With your code you'll catch some false positives

To be fair at first I tried to implement that using `grep` and not using SwiftLint but had some false positives as it didn't allow me to know enough context about the line. Then I tried a SwiftLint RegEx-based `custom_rule` but hit the limitations explained above. That's why I finally switched to an `ASTRule` to have more control over the analysis and context and be sure to annoys false positives.